scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -44,149 +49,233 @@ set.seed(12345)
 
 ### Input and output
 
-```{r}
-### Input files
+As mentioned, our input file here is a single normalized and processed SCE object, stored as an `rds` file. 
+That should be all we need to read in!
+
+Our output will be a table of per-cell information, which will include the cell type assignments we have made throughout this notebook.
+We aren't planning any significant modifications of the underlying data, so we won't bother re-saving it as a new object this time.
+
+```{r filepaths, live=TRUE}
+# directory for the input data
 data_dir <- file.path("data", "PBMC-TotalSeqB", "normalized")
+
+# the input file itself
 sce_file <- file.path(data_dir, "PBMC_TotalSeqB_normalized_sce.rds")
 
-### output files
+# A directory to store outputs
 analysis_dir <- file.path("analysis", "PBMC-TotalSeqB")
+
+# check whether the output directory exists 
 if (!dir.exists(analysis_dir)){
   dir.create(analysis_dir, recursive = TRUE)
 }
 
+# output table path
 cellinfo_file <- file.path(analysis_dir, "PBMC_TotalSeqB_cellinfo.tsv")
 ```
 
 
-Read in the SingleCellExperiment
-```{r}
-sce <- readRDS(sce_file)
-```
-
+## Exploring a CITE-seq `SingleCellExperiment`
 
-Let's explore what is in the SCE already:
+Now that the preliminary setup is out of the way, we can get started. 
+First we will read in the `SingleCellExperiment` from the input file we defined earlier.
 
-```{r}
+```{r read SCE, live=TRUE}
+# read in the SCE file
+sce <- readRDS(sce_file)
+# print a summary of the SCE
 sce
 ```
+This should look similar to the SCE objects that we have seen before, containing `counts` and `logcounts` assays where each cell is a column and each row is a gene.
+We also have some of the `rowData`, `colData` and reduced dimension matrices that we have seen before.
+
+But where are the data from the ADTs?
+We wouldn't necessarily want those stored in the main data matrices, as the characteristics of ADT barcode data is going to be quite different from gene expression data.
 
-Notice the `altExpNames`?
-Those are the ADT data, stored as an _Alternative Experiment_.
-To access these, we can use the `altExp()` function. 
+To keep the ADT data separate, we have split this data off into an _alternative experiment_ (`altExp`) slot.
+You can see the name of this `altExp` on the line `altExpNames` above. 
+We _could_ have more than one type of alternative experiment, but in this case, just the one.
+
+To access the contents of the `altExp` slot, we can use the `altExp()` function.
 Let's look at what we have in that slot:
 
-```{r}
+```{r view altExp, live=TRUE}
 altExp(sce, "ADT")
```

Comment:
I think I would just add a comment above this line that you are printing out the alternative experiment by providing the name of the experiment